atomic_reactor/plugins/pre_add_filesystem.py

```diff
@@ -358,21 +358,47 @@ def run_image_task(self, image_build_conf):
     def stream_filesystem(self, task_id, filesystem_regex):
         filesystem = self.download_filesystem(task_id, filesystem_regex)
 
-        new_base_image = self.import_base_image(filesystem)
-        self.workflow.builder.set_base_image(new_base_image)
-        defer_removal(self.workflow, new_base_image)
+        new_parent_image = self.import_base_image(filesystem)
+        new_imagename = ImageName.parse(new_parent_image)
 
-        return new_base_image
+        if self.workflow.builder.custom_base_image:
+            self.workflow.builder.set_base_image(new_imagename)
```

Comment:
Feels like we should know whether we need to call set_base_image without referring to custom_base_image.

Reply:
we have to call set_base_image only when custom base image is the last parent, in other cases when custom base image isn't last, we want to change just parent_images